internal/cmd/cmd_test.go

```diff
@@ -0,0 +1,48 @@
+package cmd
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/users"
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+// Test if a CMD can start and stop their own server when the primary server isn't being run.
+// Since all CMDs use the same infrastructure, we only need to test it for one CMD.
+func TestCMDStartsServer(test *testing.T) {
+	defer config.UNIT_TESTING_MODE.Set(config.UNIT_TESTING_MODE.Get())
+	config.UNIT_TESTING_MODE.Set(true)
+
+	defer util.RemoveRecordedTempDirs()
+
+	expectedSubstrings := []struct {
+		expectedSubstring string
+	}{
+		{`API Server Started.`},
+		{`Unix Socket Server Started.`},
+		{`API Server Stopped.`},
+		{`Unix Socket Server Stopped.`},
+	}
+
+	_, stderr, exitCode, err := RunCMDTest(test, userList, []string{}, log.GetTextLevel())
+	if err != nil {
+		test.Errorf("CMD run returned an error: '%v'.", err)
+	}
+
+	if exitCode != 0 {
+		test.Errorf("Unexpected exit code. Expected: '0', Actual: '%d'.", exitCode)
+	}
+
+	for _, testCase := range expectedSubstrings {
+		if !strings.Contains(stderr, testCase.expectedSubstring) {
+			test.Errorf("Expected substring '%s' not found in stderr: '%s'.", testCase.expectedSubstring, stderr)
+		}
+	}
+}
+
+func userList() {
+	MustHandleCMDRequestAndExit(`users/list`, users.ListRequest{}, users.ListResponse{})
+}
```

Comment:
That was my initial thought but go doesn't let you import main packages if I wanted to use that shared function here. I suppose I could have main() in cmd/users-list/main.go call a function from a different package that does everything main is doing so that function would be shared, but that doesn't follow the pattern for the rest of the CMDs.